src/System Application/App/User Permissions/src/UserPermissionsImpl.Codeunit.al

```diff
@@ -236,14 +236,16 @@ codeunit 153 "User Permissions Impl."
 
     procedure HasUserPermissionSetAssigned(UserSecurityId: Guid; Company: Text; RoleId: Code[20]; ItemScope: Option; AppId: Guid): Boolean
     var
+        User: Record User;
         NavUserAccountHelper: DotNet NavUserAccountHelper;
         Skip: Boolean;
     begin
         if HasUserPermissionSetDirectlyAssigned(UserSecurityId, Company, RoleId, ItemScope, AppId) then
             exit(true);
 
-        OnHasUserPermissionSetAssigned(Skip);
-        if Skip then
+        // NavUserAccountHelper doesn't work with buffered inserts.
```

Comment:
Do you mean bulk inserts?
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/administration/optimize-sql-bulk-inserts